Fix the automation pipeline and ingest output results

littlethumb123 opened the pull request:
## Changes Made

### Backend (`scenarios/opencaptchaworld/opencaptchaworld_judge.py`)
- **Enhanced `api_save_puzzle_data` endpoint**:
  - Now captures `user_answer` from frontend requests
  - Fetches `correct_answer` from ground truth data
  - Validates user answer using existing `check_answer()` logic
  - Adds `is_correct` boolean flag to output
  - Includes timing data (`elapsed_time`, `timestamp`)
  - **Bug fix**: Changed `json.dump(data, f)` to `json.dump(output_data, f)` to save constructed output instead of raw request

### Frontend (`assets/opencaptchaworld/static/js/script2.js`)
- **Enhanced `printAndNextPuzzle` function**:
  - Collects user answers based on puzzle input type (click, multiselect, rotation, slide, etc.)
  - Calculates elapsed time from puzzle load to submission
  - Sends comprehensive data payload to backend including user responses

## Output Format
Test results now include complete evaluation data:

{
  "puzzle_type": "Bingo",
  "puzzle_id": "bingo1.png",
  "prompt": "Please click two images to exchange their position...",
  "input_type": "bingo_swap",
  "image_path": "/captcha_data/Bingo/bingo1.png",
  "answer": [0, 7],
  "elapsed_time": 5.234,
  "timestamp": "2025-12-01T12:34:56.789Z"
}

## Test Coverage Status

### ✅ Completed Testing (16/20 puzzle types)
- Bingo
- Click_Order
- Connect_icon
- Coordinates
- Dart_Count
- Dice_Count
- Hold_Button
- Image_Matching
- Image_Recognition
- Misleading_Click
- Object_Match
- Patch_Select
- Pick_Area
- Place_Dot
- Select_Animal
- Slide_Puzzle

### ❌ Not Yet Tested (Due to file name mismatch, fixing)
- Geometry_Click
- Path_Finder
- Rotation_Match
- Unusual_Detection

gmsh committed:
Improve judge/solver guidance and feedback in OpenCaptchaWorld (#2)

- make the judge to wrap each puzzle request with clear instructions, carry forward the previous answer’s feedback, and send a final “wrap-up” message per puzzle type; evaluate_puzzle now returns both the attempt and the formatted feedback so callers can keep the solver informed
- add response-validation, feedback-generation, and fire-and-forget feedback helpers so we can gracefully tell solvers when their JSON was invalid versus when the judge hit an error, while still recording each attempt as incorrect to keep the run moving
- update the pseudo solver to ingest richer instruction messages, extract URLs via regex, ignore feedback-only messages, and log both inbound instructions and outbound JSON for easier debugging

gmsh committed:
Improve solver feedback chaining and session summary output (#3)

- keep previous_feedback alive across puzzle types so every instruction message includes the prior puzzle’s feedback, giving the solver a seamless conversation from start to finish
- replace the per-type follow-up with a single _send_final_feedback call after the full run; it now accepts the last feedback (if any) plus all_attempts, computes submission statistics, and sends a combined “final feedback + session summary” message so solvers always know how the session ended
- ensure _send_final_feedback is invoked whenever any puzzles were attempted, even if the final puzzle never produced feedback (e.g., due to network failures), so we still deliver the session recap

littlethumb123 committed:
update

littlethumb123 committed:
update

littlethumb123 committed:
Merge remote-tracking branch 'upstream/tester'

littlethumb123 committed:
add output

gmsh committed:
Fix image display properties to preserve aspect ratio and remove max-height constraint for Click Order puzzles (#4)

littlethumb123 committed:
Merge branch 'gmsh:main' into main

littlethumb123 committed:
automate check

littlethumb123 committed:
Merge branch 'main' of https://github.com/littlethumb123/agentified-captcha-benchmark-demo

hahashuo reviewed (changes requested):
1. You cannot add gitignore in gitignore
2. Please remove output folder and pseudo_purple_data folder from your git commit, 
3. Please don't change scenario.toml to use python from venv by default as venv is not required in README.md.

gmsh committed:
Fix JSON RPC error. Block access to ground truth file. Clean unused code.

- Enhance feedback handling by sending acknowledgment for feedback-only messages to fix the JSON RPC error.

- Block access to ground truth file.

- Clean unused code.

gmsh committed:
fix patch selection grid (#7)

I have fixed the Patch_Select implementation in assets/opencaptchaworld/static/js/script2.js. The issue was that it was using a generic multiselect handler instead of the specialized patch select logic from the reference implementation. I've updated the code to correctly align the image with the grid and prevent the "Download Result" button overlap.

gmsh committed:
fix the issue of incorrect component size in slide puzzles (#8)

gmsh committed:
Click Order UX Updates: add persistent numbered markers and a reset-click button (#9)

Click Order UX Updates

- assets/opencaptchaworld/static/js/script2.js now keeps dedicated click-order state, wires the Reset Clicks button, draws persistent numbered markers for each click, and limits button visibility/behavior to Click_Order puzzles so other puzzle types stay untouched.
- assets/opencaptchaworld/static/css/style.css styles the Download/Reset buttons and adds the red circular marker with a white number centered above each recorded click.
- assets/opencaptchaworld/templates/index2.html includes the Reset Clicks button in the base layout so it sits beneath Download Result and is ready for the Click_Order-specific toggle logic.

gmsh committed:
Enhance answer validation: support multiple areas for geometry click puzzles (fix puzzle geetest_000001) (#11)

gmsh committed:
Add additional answer option for bingo11 puzzle (#12)

littlethumb123 committed:
Merge branch 'gmsh:main' into main

xgong12 committed:
Update ground truth for multiple bingo puzzles (#13)

gmsh committed:
Fix patch select 2, and #16, #17, #18 and dart_puzzle_9 (#19) (#21)

* Add additional answer option for bingo11 puzzle

* Add support for optional patches in Patch_Select validation

* fix typos

* fix #17: Slide_Puzzle: "slide29.png" ground truth answer not accurate

* fix #18: Unusual_Detection: "unusual22.png" gound truth not correct. Correct answer is [2]

* fix #18: Unusual_Detection: "unusual22.png" gound truth not correct. Correct answer is [2]

* Add support for multiple correct options in Dart_Count puzzles; fix dart_puzzle_9

xgong12 committed:
Add Optional Patches to the Ground Truth for Multiple Patch_Select Puzzles (#20)

Identified Issue
For the `Patch_Select` puzzle type, we observe that some patches contain the target object but were not included in the ground truth in the original OpenCaptchaWorld paper. Visual labeling can be subjective, and disagreements between human annotators are sometimes unavoidable.

Resolution
PR #19 introduced "Optional Patches" in the original answer set for `Patch_Select` to address this issue. This PR further refines the definition and usage of optional patches to better handle incomplete or ambiguous human annotations.

Optional patches cover two scenarios:

1. Patches that contain the target object but were missed in the original OpenCaptchaWorld paper ground truth annotations.
2. Patches where visual evidence is ambiguous and reasonable annotators may disagree.

The goal is to preserve the benchmark’s ability to distinguish agent performance while allowing reasonable flexibility in cases of ambiguous human annotation. 

Example (`Image2`)
Patch 17 contains a visible butterfly segment but was not labeled in the original annotation. It is therefore marked as *optional* to avoid penalizing agents that correctly identify it. In contrast, patches 7 and 12 contain clearly identifiable butterfly regions and are excluded from the optional set to maintain evaluation strictness.

Co-authored-by: Maosheng Guo <gmsh@users.noreply.github.com>

littlethumb123 committed:
Merge branch 'gmsh:main' into main

gmsh commented:
closing this PR as it's no longer intended to be merged.
